src/vs/workbench/api/node/extHostWorkspace.ts

```diff
@@ -78,6 +78,18 @@ export class ExtHostWorkspace implements ExtHostWorkspaceShape {
 		}
 	}
 
+	addWorkspaceFolder(uri: URI, name?: string): Thenable<void> {
+		return this._proxy.$addFolder(uri, name);
+	}
+
+	removeWorkspaceFolder(folder: vscode.WorkspaceFolder): Thenable<void> {
+		if (this.getWorkspaceFolders().indexOf(folder) === -1) {
+			return Promise.resolve(undefined);
```

Comment:
Maybe make this return a boolean to indicate success/failure. Might also be useful when having a user-confirmation in between. Then `false` could also mean blocked by the user

Reply:
@jrieken yeah added via https://github.com/Microsoft/vscode/pull/36820/commits/1dc821b614609990436008b2aff5fd262a343b09